### PR TITLE
New Tab Page WebUI - remove widget visibility migration

### PR DESCRIPTION
**Pull request by petemill**
Resolves https://github.com/brave/brave-browser/issues/20501

The migration was also checking for whether the user had previously connected to a widget by requesting data, instead of checking the token.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by bbondy — APPROVED** [review]

File: browser/brave_profile_prefs.cc [hunk 1]
```diff
@@ -317,7 +317,6 @@ void RegisterProfilePrefs(user_prefs::PrefRegistrySyncable* registry) {
   registry->RegisterBooleanPref(kNewTabPageShowBraveTalk, false);
   registry->RegisterBooleanPref(kNewTabPageShowGemini, true);
   registry->RegisterBooleanPref(kNewTabPageHideAllWidgets, false);
-  registry->RegisterBooleanPref(kNewTabPageWidgetVisibilityMigrated, false);
```
Comment (simonhong) [inline browser/brave_profile_prefs.cc:320]: How about changing all crypto widget's visibility default value to `false` if we still want to hide them all by default?
I think most of users have migrated prefs. So, I think it's fine to set `false` all.

**Comment by wknapik** [comment]
@petemill I just added the `CI/run-network-audit` label. You might want to rebuild.

File: browser/brave_profile_prefs.cc (hunk 1 above)
Comment (petemill) [reply browser/brave_profile_prefs.cc:320]: Good idea, that keeps the logic the same as the JS migration - for new users they are default to hidden, and existing users would already have a value. I've pushed this. Please take a look.

**Review by simonhong — APPROVED** [review]
++

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build:

```
Brave | 1.36.48 Chromium: 97.0.4692.71 (Official Build) nightly (64-bit)
-- | --
Revision | adefa7837d02a07a604c1e6eff0b3a09422ab88d-refs/branch-heads/4692@{#1247}
OS | Windows 11 Version 21H2 (Build 22000.434)
```

**`Clean Install Cases`**

Verified the STR/Cases outlined via https://github.com/brave/brave-browser/issues/20501#issue-1102759124 and ensured that there wasn't any outbound network connections when opening the NTP on a new profile. Example:

startupNetworkCalls

Also ensured that none of the crypto widgets were visible on a new profile but were available  via the following:
* `Customise dashboard` -> `Cards`

**`Upgrade Cases`**

For the following cases, upgrade from `1.36.47 Chromium: 97.0.4692.71` --> `1.36.48 Chromium: 97.0.4692.71`

* ensured that crypto widgets are being displayed in the widget stack if a user has already enabled them
  * checked using Binance, FTX & Gemini
 * ensured that users stay authenticated after upgrading when the widget is active/visible under the widget stack
 * ensured that users stay authenticated after upgrading when the authenticated widget isn't active/visible in the widget stack
 * ensured that the correct order/active widget is maintained after upgrading
 * ensured that crypto widgets are not being removed from the widget stack and into `Customise dashboard` -> `Cards` when they've been enabled

**Pull request merged**